### PR TITLE
Using suggested polyfill for checkVisibility

### PR DESCRIPTION
**Pull request by jonrohan**
_Authors: Please fill out this form carefully and completely._

_Reviewers: By approving this Pull Request you are approving the code change, as well as its deployment and mitigation plans._
_Please read this description carefully. If you feel there is anything unclear or missing, please ask for updates._

### What are you trying to accomplish?

@keithamus suggested this polyfill after I merged https://github.com/primer/view_components/pull/2292#discussion_r1364664773

### CODE REVIEW

**Review by camertron — APPROVED** [review]
Do we have test coverage for the visibility stuff?

**Comment by jonrohan** [comment]
> Do we have test coverage for the visibility stuff?

We have some system tests that cover it via focus states

File: app/components/primer/alpha/action_bar_element.ts [hunk 1]
```diff
@@ -86,16 +86,27 @@ class ActionBarElement extends HTMLElement {
...
   }
 
-  #isVisible(element: HTMLElement): boolean {
-    // Safari doesn't support `checkVisibility` yet.
-    if (typeof element.checkVisibility === 'function') return element.checkVisibility()
-
-    return Boolean(element.offsetParent || element.offsetWidth || element.offsetHeight)
+  #checkVisibility(el: HTMLElement, {checkOpacity = false, checkVisibilityCSS = false} = {}): boolean {
+    if (!el) return false
+    if (typeof el.checkVisibility === 'function') {
```
Comment (jonrohan) [inline app/components/primer/alpha/action_bar_element.ts:96]: @keithamus Is there some typescript config that needs updating for these errors?

image

**Comment by keithamus** [comment]
@jonrohan I've added `checkVisibility` to the github `browser-support` package. We should probably include that in this lib, or maybe just make a note that it should be required?

See https://github.com/github/browser-support/pull/47 for more.

**Comment by jonrohan** [comment]
I'm going to close this then and look at adding the browser-support package.

**Pull request closed without merging**